lua/starfall/libs_sv/navmesh.lua

```diff
@@ -0,0 +1,780 @@
+-- Global to all starfalls
+local checkluatype = SF.CheckLuaType
+local registerprivilege = SF.Permissions.registerPrivilege
+
+-- Register privileges
+-- 3 = no one, 2 = anyone, 1 = admin
+registerprivilege("navmesh.generate", "Begin generation", "Allows the user to generate a navmesh for the map. This process is highly resource intensive and not wise to use during normal gameplay", { usergroups = { default = 3 } })
+registerprivilege("navmesh.reset", "Reset navmesh", "Allows the user to reset the map navmesh. You shouldn't enable this.", { usergroups = { default = 3 } })
+registerprivilege("navmesh.load", "Load navmesh", "Allows the user to load and discard changes on the current map navmesh. You shouldn't enable this.", { usergroups = { default = 3 } })
+registerprivilege("navmesh.save", "Reset navmesh", "Allows the user to save the map navmesh. You shouldn't enable this.", { usergroups = { default = 3 } })
+
+registerprivilege("navmesh.modify", "Modify navmesh", "Allows the user to modify the map navmesh before generation", { usergroups = { default = 3 } })
+
+registerprivilege("navarea.create", "Create NavArea", "Allows the user to create a CNavArea", { usergroups = { default = 2 } })
+registerprivilege("navarea.openlist", "Modify NavArea Openlist", "Allows the user to modify the global navarea openlist", { usergroups = { default = 1 } })
+
+--- Library for navmesh navigation with the NavArea type
+-- @name navmesh
+-- @class library
+-- @libtbl navmesh_library
+SF.RegisterLibrary("navmesh")
+
+--- NavArea type, returned by navmesh library functions
+-- @name NavArea
+-- @class type
+-- @libtbl navarea_methods
+-- @libtbl navarea_meta
+SF.RegisterType("NavArea", true, false, debug.getregistry().NavArea, "LockedNavArea")
```

Comment:
Probably don't need to register the metatable `debug.getregistry().NavArea` since this type won't ever be used with WrapObject/UnwrapObject